bridges/NineGagBridge.php

```diff
@@ -0,0 +1,330 @@
+				'name' => 'Section',
+				'type' => 'list',
+				'required' => true,
+				'values' => array(
+					'Animals' => 'cute',
+					'Anime & Manga' => 'anime-manga',
+					'Ask 9GAG' => 'ask9gag',
+					'Awesome' => 'awesome',
+					'Basketball' => 'basketball',
+					'Car' => 'car',
+					'Classical Art Memes' => 'classicalartmemes',
+					'Comic' => 'comic',
+					'Cosplay' => 'cosplay',
+					'Countryballs' => 'country',
+					'DIY & Crafts' => 'imadedis',
+					'Drawing & Illustration' => 'drawing',
+					'Fan Art' => 'animefanart',
+					'Food & Drinks' => 'food',
+					'Football' => 'football',
+					'Fortnite' => 'fortnite',
+					'Funny' => 'funny',
+					'GIF' => 'gif',
+					'Gaming' => 'gaming',
+					'Girl' => 'girl',
+					'Girly Things' => 'girly',
+					'Guy' => 'guy',
+					'History' => 'history',
+					'Home Design' => 'home',
+					'Horror' => 'horror',
+					'K-Pop' => 'kpop',
+					'LEGO' => 'lego',
+					'League of Legends' => 'leagueoflegends',
+					'Movie & TV' => 'movie-tv',
+					'Music' => 'music',
+					'NFK - Not For Kids' => 'nsfw',
+					'Overwatch' => 'overwatch',
+					'PC Master Race' => 'pcmr',
+					'PUBG' => 'pubg',
+					'Pic Of The Day' => 'photography',
+					'Pokémon' => 'pokemon',
+					'Politics' => 'politics',
+					'Relationship' => 'relationship',
+					'Roast Me' => 'roastme',
+					'Satisfying' => 'satisfying',
+					'Savage' => 'savage',
+					'School' => 'school',
+					'Sci-Tech' => 'science',
+					'Sport' => 'sport',
+					'Star Wars' => 'starwars',
+					'Superhero' => 'superhero',
+					'Surreal Memes' => 'surrealmemes',
+					'Timely' => 'timely',
+					'Travel' => 'travel',
+					'Video' => 'video',
+					'WTF' => 'wtf',
+					'Wallpaper' => 'wallpaper',
+					'Warhammer' => 'warhammer',
+				),
+			),
+			't' => array(
+				'name' => 'Type',
+				'type' => 'list',
+				'required' => true,
+				'values' => array(
+					'Hot' => 'hot',
+					'Fresh' => 'fresh',
+				),
+			),
+			'p' => array(
+				'name' => 'Pages',
+				'type' => 'number',
+				'defaultValue' => 3,
+			),
+		),
+	);
+
+	const MIN_NBR_PAGE = 1;
+	const MAX_NBR_PAGE = 6;
+
+	protected $p = null;
+
+	public function collectData() {
+		$url = sprintf(
+			'%sv1/group-posts/group/%s/type/%s?',
+			self::URI,
+			$this->getGroup(),
+			$this->getType()
+		);
+		$cursor = 'c=10';
+		$posts = array();
+		for ($i = 0; $i < $this->getPages(); ++$i) {
+			$content = getContents($url.$cursor);
+			$json = json_decode($content, true);
+			$posts = array_merge($posts, $json['data']['posts']);
+			$cursor = $json['data']['nextCursor'];
+		}
+
+		foreach ($posts as $post) {
+			$item['uri'] = $post['url'];
+			$item['title'] = $post['title'];
+			$item['content'] = self::getContent($post);
+			$item['categories'] = self::getCategories($post);
+			$item['timestamp'] = self::getTimestamp($post);
+
+			$this->items[] = $item;
+		}
+	}
+
+	public function getName() {
+		if ($_GET['bridge'] === substr(get_class($this), 0, -6)) {
```

Comment:
This is not necessary, this function is only called when the bridge is accessed

Reply:
I had to do this because on the bridges page, I had the name of the bridge that changed, but I changed the function to change the name only if I have the parameters.

Reply:
Sorry that wasn't clear: It would be better not accessing these variables directly. For example, RSS-Bridge is also accessible by CLI, which would not work.
You can use ```$this->queriedContext``` instead.